google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PublisherImpl.java

```diff
@@ -90,55 +84,42 @@
   private final Duration sendBundleDeadline;
   private ScheduledFuture<?> currentAlarmFuture;
 
-  PublisherImpl(Builder builder) throws IOException {
-    topic = builder.topic;
-
-    maxBundleMessages = builder.bundlingSettings.getElementCountThreshold();
-    maxBundleBytes = builder.bundlingSettings.getRequestByteThreshold();
-    maxBundleDuration = builder.bundlingSettings.getDelayThreshold();
+  PublisherImpl(String topic, PubSubOptions psOptions, Settings settings) throws IOException {
```

Comment:
Don't abbreviate the "ps" here - I would prefer spelled out, or maybe just `options`.

Bigger picture: It is very confusing to have both options and settings.